### ISSUE TITLE
Lot of troubles with peer dependencies

### ISSUE THREAD

**Issue by zwhitchcox** [transitive-bullshit/create-react-library#181]
I have `react` as a peer dependency, but whenever I imported the package into another library, it said "mismatched versions of react". So, I went in a deleted `react` from `node_modules`, and that seemed to work.

I thought  the peer dependencies library was supposed to take care of that though.

But that has been working fine, I just manually delete `react` from `node_modules` any time something else I add to my dependencies imports it.

Only now, I need `@material-ui/core` as a peer dependency, but whenever I try to use it, there are 2 copies of it. This is a problem, because the conflicting copies mess up the colors of the other one, and obviously it adds to my bundle size and will probably cause other unforeseen problems as well.

I'm just wondering what the proper way to handle peer dependencies is. I add them to my peer dependencies, but it doesn't seem to work. Any suggestions? I [opened an issue](https://github.com/pmowrer/rollup-plugin-peer-deps-external/issues/18) with the rollup-plugin-peer-deps-external package creator, but he has not responded, and it looks like they basically never respond to any issues.

**Comment by zwhitchcox** [comment]
Oh, I just realized you don't respond to issues either. Great.

**Comment by zwhitchcox** [comment]
Guess I have to use something else.

**Comment by transitive-bullshit** [comment]
Sorry, Zane. I would honestly really love to help you and other CRL users out but life is busy other other priorities.

If you're using TypeScript, I'd highly recommend using [tsdx](https://github.com/palmerhq/tsdx). Otherwise, I think [microbundle](https://github.com/developit/microbundle) will be very useful to you.

Best of luck.

**Comment by zwhitchcox** [comment]
Hey, thanks for your response!

I actually figured it out last night around 1AM and didn't have the energy to close this issue though.

But the problem was in the `typescript` roll-up plugin, not in the roll-up bundling itself.

All I had to do was add the line

`declare module "@material-ui/core";`

to the `typings.d.ts` file.

Hopefully this will help anyone else who finds runs into this. I don't really understand why this works, but it works.

Thanks for creating the project, I am really enjoying it most of the time.
